regression/contracts/quantifiers-exists-ensures-01/main.c

```diff
@@ -0,0 +1,18 @@
+int f1(int *arr) __CPROVER_ensures(__CPROVER_exists {
+  int i;
+  (0 <= i && i < 10) && arr[i] == i
+})
```

Comment:
Why not `(0 <= i && i < 10) ==> arr[i] == i`?

Reply:
Yes. `forall i . (0 <= i && i < 10) && arr[i] == i` is just false. It should be an implication.

Reply:
I am using `(0 <= i && 1 < 10) && arr[i] == i` within an existential quantifier. I believe that it is correct as is.

Reply:
The correctness would depend on what we want to check about the function. Is it:
1. If there is some value within the range then the property holds, or
2. There must be a value within the range _and_ the property should hold

Consider this example:
```
exists x : char . 1000 <= (int) x <= 5000 ==> pred(x)
```
This would evaluate to `true` and models (1) correctly because there is no char between `1000` and `5000` in the first place.

```
exists x : char . 1000 <= (int) x <= 5000 && pred(x)
```
This would evaluate to `false` and models (2) correctly.

Reply:
@ArenBabikian your original change above was fine in this case.

The proper way to handle the emptiness case would be to add an implication for non-emptiness:
https://github.com/diffblue/cbmc/blob/66d9d175919787fb08107fe4d28759f284be8f46/regression/contracts/quantifiers-loop-03/main.c#L17-L24

Sorry about the confusion; fixing this (among other issues) in #6145.